Content.Server/Nyanotrasen/Abilities/Borgs/RandomizedCandySystem.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Linq;
+using Content.Server.Nutrition.Components;
+using Content.Shared.Abilities.Borgs;
+using Content.Shared.Nutrition;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Abilities.Borgs;
+
+/// <summary>
+/// Gives things with a <see cref="RandomizedCandyComponent"/> a random flavor, with corresponding appearance and
+/// examine text.
+/// </summary>
+public sealed partial class RandomizedCandySystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly MetaDataSystem _metaData = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+
+    /// <summary>
+    /// Flavors that are masked by the candy.
+    /// </summary>
+    private static readonly string[] MaskedReagents = { "Sugar", "Iron" }; // sugar is obvious and iron is "metallic" :(
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RandomizedCandyComponent, MapInitEvent>(OnInit);
+    }
+
+    private void OnInit(EntityUid uid, RandomizedCandyComponent candyComp, MapInitEvent args)
+    {
+        // pick a random flavor
+        var flavors = _prototypeManager.EnumeratePrototypes<CandyFlavorPrototype>();
+        var candyFlavor = _random.Pick(flavors.ToList());
+
+        // color the candy :3
+        if (TryComp(uid, out AppearanceComponent? appearance))
+            _appearance.SetData(uid, RandomizedCandyVisuals.Color, candyFlavor.Color, appearance);
```

Comment:
dont need the trycomp SetData already has a resolve, just remove the appearance arg at the end